### PR TITLE
[ntuple] Speed up reading/writing of simple vectors

### PR DESCRIPTION
**Pull request by jblomer**
For RVecs of simple types, do a vector read of the item's principal columns instead of reading items one by one.

### CODE REVIEW

File: tree/ntuple/v7/src/RNTupleDescriptorFmt.cxx [hunk 1]
```diff
@@ -110,6 +110,9 @@ void ROOT::Experimental::RNTupleDescriptor::PrintInfo(std::ostream &output) cons
       info.fElementSize = elementSize;
       info.fType = column.second.GetModel().GetType();
 
+      if (column.second.IsAliasColumn())
+         continue;
```
Comment (vepadulano) [inline tree/ntuple/v7/src/RNTupleDescriptorFmt.cxx:114]: It is not straightforward to understand why we skip alias columns.

File: tree/ntuple/v7/src/RField.cxx [hunk 2]
```diff
@@ -2006,6 +2006,11 @@ void ROOT::Experimental::RRVecField::ReadGlobalImpl(NTupleSize_t globalIndex, vo
          CallGenerateValueOn(*fSubFields[0], begin + (i * fItemSize));
    }
 
+   if (fSubFields[0]->IsSimple()) {
+      GetPrincipalColumnOf(*fSubFields[0])->ReadV(collectionStart, nItems, begin);
+      return;
```
Comment (vepadulano) [inline tree/ntuple/v7/src/RField.cxx:2011]: The change definitely makes sense. Unfortunately, I can't tell how this leads to the current failures in the test `onDiskPage && (bytesOnStorage == onDiskPage->GetSize()) violated at line 341`

File: tree/ntuple/v7/src/RNTupleDescriptorFmt.cxx (hunk 1 above)
Comment (jblomer) [reply tree/ntuple/v7/src/RNTupleDescriptorFmt.cxx:114]: Addressed in #13589 which is underlying this PR.

File: tree/ntuple/v7/src/RField.cxx (hunk 2 above)
Comment (jblomer) [reply tree/ntuple/v7/src/RField.cxx]: Found it: empty simple vectors trigger that. Should be fixed now.

**Review by vepadulano — APPROVED** [review]
Tests are green now, the changes look good, thank you!

**Pull request merged**